src/libs/antares/study/parts/short-term-storage/AdditionalConstraints.cpp

```
@@ -18,11 +18,12 @@
 ** You should have received a copy of the Mozilla Public Licence 2.0
 ** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
 */
-#include "antares/study/parts/short-term-storage/AdditionalConstraint.h"
+#include "antares/study/parts/short-term-storage/AdditionalConstraints.h"
+#include <algorithm>
 
 namespace Antares::Data::ShortTermStorage
 {
-AdditionalConstraint::ValidateResult AdditionalConstraint::validate() const
+AdditionalConstraints::ValidateResult AdditionalConstraints::validate() const
```

We don't need to define a **struct ValidateResult** : it can be replaced with a `std::pair<bool, std::string>`.

That said, instead of returning something, we could throw exception (`std::runtime_error` ?) inside **validate()**.
This would simplify the function a bit : 
- it would return **void**
- it would avoid the strange default return `{true, ""}`

Calling it would be a bit simpler and conventional : 
```C++
try
{
    additional_constraints.validate();
}
catch (const std::runtime_error& ex)
{
    logs.error() << "Invalid constraint in section: " << section->name;
    logs.error() << ex.what();
    return false;
}
```

---

pair done

---

i don't thinh that std::pair<bool, std::string> is more meaningful than struct other { bool ok; string msg;} come on!
pair<bool, string> a; a.first = true;

other o; o.ok = true;

---

I have to agree with @a-zakir on this one

---

The suggestion of using a `std::pair<bool, std::string>` instead of a **struct** was a part of this remark.
What about the other part (with the code sample) ?

---

we don't necessarily want to throw exception, the return boolean is used by the caller 
bool AreaList::**loadFromFolder** --> bool STStorageInput::**loadAdditionalConstraints** --> AdditionalConstraints::ValidateResult AdditionalConstraints::**validate**()

---

in your suggestion:
``` C++
try
{
    additional_constraints.validate();
}
catch (const std::runtime_error& ex)
{
    logs.error() << "Invalid constraint in section: " << section->name;
    logs.error() << ex.what();
    return false;
}

```

  i am doubtful with throwing an exception to catch it the line after.
  can you give an example of code in the project that use this technique?

---

Well, the error handling in **Antares Simulator** in not necessarily at the state of art.
This suggestion has the advantage of : 
- simplifying the **validate()** function, especially its return type,
- make the caller as clear as it was before, with the **try... catch...** block (which indicates we're dealing with errors)

But in the end, this point is a detail